src/compiler/checker.ts

```diff
@@ -1824,7 +1825,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             } while (toMarkSkip && toMarkSkip !== containingCall);
             getNodeLinks(containingCall).resolvedSignature = undefined;
         }
+        isInferencePartiallyBlocked = true;
         const result = fn();
+        isInferencePartiallyBlocked = false;

```

Comment:
Is this reentrant such that you should save the value and restore it, rather than unconditionally setting false?

Reply:
I intentionally ignored this to keep the implementation simpler as I dont think it is